core/inst_handlers/zcmp/RvzcmpInsts.cpp

```diff
@@ -0,0 +1,159 @@
+
+        // Store RA (1) and 0-12 saved registers (8-9, 18-27) to the stack frame
+        // Have to start at idx 1 since Mavis includes the SP in the list of sources
+        XLEN addr = READ_INT_REG<XLEN>(state, 2);
+        const XLEN new_sp_val = addr + inst->getStackAdjustment();
+        const auto & src_reg_list = inst->getMavisOpcodeInfo()->getSourceOpInfoList();
+        for (uint32_t idx = 1; idx < src_reg_list.size(); ++idx)
+        {
+            const auto src = src_reg_list.at(idx);
+            addr -= sizeof(XLEN);
+            const XLEN src_reg_val = READ_INT_REG<XLEN>(state, src.field_value);
+            state->writeMemory<XLEN>(addr, src_reg_val);
+        }
+
+        // Update stack pointer
+        WRITE_INT_REG<XLEN>(state, 2, new_sp_val);
+
+        return ++action_it;
+    }
+
+    template <typename XLEN>
+    Action::ItrType RvzcmpInsts::popHandler_(pegasus::PegasusState* state,
+                                             Action::ItrType action_it)
+    {
+        pop_<XLEN>(state);
+        return ++action_it;
+    }
+
+    template <typename XLEN>
+    Action::ItrType RvzcmpInsts::popretHandler_(pegasus::PegasusState* state,
+                                                Action::ItrType action_it)
+    {
+        pop_<XLEN>(state);
+
+        // Set PC to RA (1)
+        const XLEN ra_val = READ_INT_REG<XLEN>(state, 1);
+        state->setNextPc(ra_val);
+
+        return ++action_it;
+    }
+
+    template <typename XLEN>
+    Action::ItrType RvzcmpInsts::popretzHandler_(pegasus::PegasusState* state,
+                                                 Action::ItrType action_it)
+    {
+        pop_<XLEN>(state);
+
+        // Set PC to RA (1)
+        const XLEN ra_val = READ_INT_REG<XLEN>(state, 1);
+        state->setNextPc(ra_val);
+
+        // Set A0 (10) to zero
+        WRITE_INT_REG<XLEN>(state, 10, 0);
+
+        return ++action_it;
+    }
+
+    template <typename XLEN>
+    Action::ItrType RvzcmpInsts::mva01sHandler_(pegasus::PegasusState* state,
+                                                Action::ItrType action_it)
+    {
+        const PegasusInstPtr & inst = state->getCurrentInst();
+        const XLEN rs1_val = READ_INT_REG<XLEN>(state, inst->getRs1());
+        const XLEN rs2_val = READ_INT_REG<XLEN>(state, inst->getRs2());
+
+        // Write RS1 to A0 (10)
+        WRITE_INT_REG<XLEN>(state, 10, rs1_val);
+        // Write RS2 to A1 (11)
+        WRITE_INT_REG<XLEN>(state, 11, rs2_val);
+
+        return ++action_it;
+    }
+
+    template <typename XLEN>
+    Action::ItrType RvzcmpInsts::mvsa01Handler_(pegasus::PegasusState* state,
+                                                Action::ItrType action_it)
+    {
+        const PegasusInstPtr & inst = state->getCurrentInst();
+        const XLEN a0_val = READ_INT_REG<XLEN>(state, 10);
+        const XLEN a1_val = READ_INT_REG<XLEN>(state, 11);
+
+        // FIXME: Does Mavis handle this?
```

Comment:
I wondered the same thing.  I'm in the middle of fix this instruction right now.  I can make that illegal